Add validator exit process docs (#109)
onmax/nimiq-developer-center: build/set-up-your-own-node/becoming-a-validator.md

```

:::

## Remove your Validator

> Anyone can remove their validator when they no longer want to participate in the consensus. This process involves three transactions: `deactivate`, `retire`, and `delete` — and it may take up to 2 days to complete before the full 100'000 NIM deposit can be recovered.

Reviewer: ```suggestion
Anyone can remove their validator when they no longer want to participate in block production. This process involves three transactions: `deactivate`, `retire`, and `delete` — and it may take up to 2 days to complete before the full 100'000 NIM deposit can be recovered.
```

- `deactivate` marks the validator as inactive
- `retire` confirms the validator is leaving permanently
- `delete` removes the validator and returns the deposit  

The process may take up to 2 days due to the protocol’s requirement to allow time for reporting any potential misbehavior before the validator can be fully removed and the deposit returned.

> #### Deactivate your Validator

Reviewer: We need another section here. The title of this section can be something like "Disable automatic reactivation". The node operator can configure the node such that once deactivated, it will automatically reactivate. When you are in the process of deleting your validator, you don't want this mechanism to trigger and basically constantly reactivate the node.

In the `client.toml` ([link](https://github.com/nimiq/core-rs-albatross/blob/b1e8c0f26f55039861c1cc9a5112ad08ce87067c/lib/src/config/config_file/client.example.toml#L356)) there is a config flag that indicates whether or not the node should automatically reactivate once it detects that is has been deactivated. If set to `false`, no action need to be taken. If it is set to `true`, the operator has to call this RPC method in order to disable automatic reactivation: [link](https://www.nimiq.com/developers/build/set-up-your-own-node/rpc-docs/#setautomaticreactivation). If the node operator decides to restart his node afterwards, this change is lost and the client falls back to the value in the `client.toml`.

::: details ARPL {open}

```
validator:deactivate <validator_address> <signing_private_key>
```

:::

::: details CURL

The manual way with `curl` requires a few more parameters:
1. The address paying the transaction fee
2. The address of the validator you are deactivating
3. The signing key of the validator for validity and authorization
4. The fee for the transaction (can be zero)
5. The validity start height of the transaction (+0 means the node takes the current block height)

```bash
curl 'http://localhost:8648' -H 'Content-Type: application/json' \

>     --data-raw '{"method": "sendDeactivateValidatorTransaction", "params": ["<sender_wallet>", "<validator_address>", "<signing_secret_key>", 0, 0], "jsonrpc": "2.0", "id": 1}'

Reviewer: ```suggestion
    --data-raw '{"method": "sendDeactivateValidatorTransaction", "params": ["<sender_wallet>", "<validator_address>", "<signing_secret_key>", 0, "+0"], "jsonrpc": "2.0", "id": 1}'
```

```

:::

#### Retire you Validator

::: details ARPL {open}

```
validator:retire <validator_address>
```

:::

::: details CURL

The manual way with `curl` requires a few more parameters:
1. The address paying the transaction fee
2. The address of the validator you are retiring
3. The fee for the transaction (can be zero)
4. The validity start height of the transaction (+0 means the node takes the current block height)

```bash
curl 'http://localhost:8648' -H 'Content-Type: application/json' \

>     --data-raw '{"method": "sendRetireValidatorTransaction", "params": ["<sender_wallet>", "<validator_wallet>", 0, 0], "jsonrpc": "2.0", "id": 1}'

Reviewer: ```suggestion
    --data-raw '{"method": "sendRetireValidatorTransaction", "params": ["<sender_wallet>", "<validator_wallet>", 0, "+0"], "jsonrpc": "2.0", "id": 1}'
```

```

:::

#### Delete your Validator

::: details ARPL {open}

```
validator:delete <validator_address>
```

:::

::: details CURL

The manual way with `curl` requires a few more parameters:
1. The address of your validator wallet submitting the `delete` transaction
2. The address where the funds will be sent
3. The amount in NIM to withdraw
4. The fee for the transaction (can be zero)
5. The validity start height of the transaction (+0 means the node takes the current block height)

```bash
curl 'http://localhost:8648' -H 'Content-Type: application/json' \

>     --data-raw '{"method": "sendDeleteValidatorTransaction", "params": ["<validator_wallet>", "<recipient>", 0, 0, 0], "jsonrpc": "2.0", "id": 1}'

Reviewer: ```suggestion
    --data-raw '{"method": "sendDeleteValidatorTransaction", "params": ["<validator_wallet>", "<recipient>", 0, 0, "+0"], "jsonrpc": "2.0", "id": 1}'
```
